cmd/signozcollectormigrator/migrate.go

```diff
@@ -0,0 +1,99 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"os"
+
+	migrationmanager "github.com/SigNoz/signoz-otel-collector/migrationManager"
+	"github.com/spf13/pflag"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+)
+
+func init() {
+	// init zap logger
+	config := zap.NewProductionConfig()
+	config.EncoderConfig.EncodeLevel = zapcore.LowercaseLevelEncoder
+	config.EncoderConfig.TimeKey = "timestamp"
+	config.EncoderConfig.EncodeTime = zapcore.ISO8601TimeEncoder
+	logger, err := config.Build()
+	if err != nil {
+		log.Fatalf("Failed to initialize zap logger %v", err)
+	}
+	// replace global logger
+	zap.ReplaceGlobals(logger)
+}
+
+func main() {
+	logger := zap.L().With(zap.String("component", "migrate cli"))
+	f := pflag.NewFlagSet("Collector Migrator CLI Options", pflag.ExitOnError)
+
+	f.Usage = func() {
+		fmt.Println(f.FlagUsages())
+		os.Exit(0)
+	}
+
+	f.String("dsn", "", "Clickhouse DSN")
+	f.String("cluster-name", "cluster", "Cluster name to use while running migrations")
+	f.Bool("multi-node-cluster", false, "True if the dsn points to a multi node clickhouse cluster, false otherwise. Defaults to false.")
+	f.Bool("disable-duration-sort-feature", false, "Flag to disable the duration sort feature. Defaults to false.")
+	f.Bool("disable-timestamp-sort-feature", false, "Flag to disable the timestamp sort feature. Defaults to false.")
+
+	err := f.Parse(os.Args[1:])
+	if err != nil {
+		logger.Fatal("Failed to parse args", zap.Error(err))
+	}
+
+	dsn, err := f.GetString("dsn")
+	if err != nil {
+		logger.Fatal("Failed to get dsn from args", zap.Error(err))
+	}
+
+	clusterName, err := f.GetString("cluster-name")
+	if err != nil {
+		logger.Fatal("Failed to get cluster name from args", zap.Error(err))
+	}
+
+	multiNodeCluster, err := f.GetBool("multi-node-cluster")
+	if err != nil {
+		logger.Fatal("Failed to get multi node cluster flag from args", zap.Error(err))
+	}
+
+	disableDurationSortFeature, err := f.GetBool("disable-duration-sort-feature")
+	if err != nil {
+		logger.Fatal("Failed to get disable duration sort feature flag from args", zap.Error(err))
+	}
+
+	disableTimestampSortFeature, err := f.GetBool("disable-timestamp-sort-feature")
+	if err != nil {
+		logger.Fatal("Failed to get disable timestamp sort feature flag from args", zap.Error(err))
+	}
+
+	if dsn == "" {
+		logger.Fatal("dsn is a required field")
+	}
+
+	// set cluster env so that golang-migrate can use it
+	// the value of this env would replace all occurences of {{.SIGNOZ_CLUSTER}} in the migration files
+	os.Setenv("SIGNOZ_CLUSTER", clusterName)
+
+	managerConfig := migrationmanager.NewConfig(
```

Comment:
(nit): common practice is to expose the config options and let the module consumer create the config object directly instead of the `NewConfig` func.

Reply:
Yeah i agree @srikanthccv i had used the same thing earlier but then i had to create a manager config as the parameters were increasing
I wanted to create a singular struct with the config and wanted to use it both in the manager and the individual migrators for metrics, etc..there were couple of things i was trying to do via the newConfig func - 
- i wanted the manager config to be a superset of the migrator config(for being able to easily add usecases like running migrations only for one of logs/metrics/traces)
- i wanted to keep cmd's migrate.go only dependent on the migrationmanager's manager.go file.Didn't want to import migrator config inside the migrate.go file
But looking back the number of arguments to New reaching 5 was kind of my threshold for adding a separate config struct for manager..now that we have removed the multi node flag, it brings the number of arguements to the New func to 4 which makes this whole config struct look like an overkill to me
For now, i have removed the manager's Config struct and am passing 4 arguments to manager's new func to create a new manager

Reply:
You misunderstood my point, it's fine to have Config objects and have the modules initialising them. But the `NewConfig` function to create a `Config` object is uncommon.